can-define-class.js

```diff
@@ -88,15 +90,39 @@ function define(Base = Object) {
 
 		constructor(props) {
 			super();
-			Object.defineProperty(this, inSetupSymbol, {
-				configurable: true,
-				enumerable: false,
-				value: true,
-				writable: true
-			});
-			new.target._initDefines();
-			setup.call(this, props, new.target.seal);
-			this[inSetupSymbol] = false;
+			if (this instanceof Element) {
+				this[constructorPropsSymbol] = props;
+			} else {
+				this.setup(props);
+			}
+		}
+
+		connectedCallback() {
+			if (typeof super.connectedCallback === "function") {
+				super.connectedCallback();
+			}
+			this.setup( this[constructorPropsSymbol] );
+		}
+
+		setup(props) {
+			if(!this[hasBeenSetupSymbol]) {
+				Object.defineProperty(this, inSetupSymbol, {
```

Comment:
I think there are a few defineProperty() calls in this file now, what do you think about adding a helper function so we can call `setNonEnumerable(obj, symbol, value)` or something?